kivy/base.py

```diff
@@ -491,9 +491,8 @@ def runTouchApp(widget=None, slave=False):
 
 def stopTouchApp():
     '''Stop the current application by leaving the main loop'''
-    if EventLoop is None:
-        return
-    if EventLoop.status != 'started':
+    if EventLoop is None or EventLoop.status != 'started':
+        Clock.schedule_once(lambda dt: stopTouchApp(), 0)
```

Comment:
This seems dangerous. Imagine you want to stop and start again the event loop (which is not really support now, but still...), calling stop too late after it's already stopping will keep scheduling to stop until the next time you run it, which is not what was wanted. I think we should only schedule if we know that the event loop is starting, but not before it started at all and not if it's already stopping.